common/test/src/IO/WorldReaderTest.cpp

```diff
@@ -1076,6 +1077,58 @@ namespace TrenchBroom {
             CHECK(mySubGroup->childCount() == 1u);
         }
 
+        TEST_CASE("WorldReaderTest.parseLayersAndGroupsAndRetainIds", "[WorldReaderTest]") {
+            const std::string data(R"(
+{
+"classname" "worldspawn"
+}
+{
+"classname" "func_group"
+"_tb_type" "_tb_layer"
+"_tb_name" "Layer"
+"_tb_id" "7"
+}
+{
+"classname" "func_group"
+"_tb_type" "_tb_group"
+"_tb_name" "Group 1"
+"_tb_id" "7"
+"_tb_layer" "7"
+}
+{
+"classname" "func_group"
+"_tb_type" "_tb_group"
+"_tb_name" "Group 2"
+"_tb_id" "22"
+}
+)");
+            const vm::bbox3 worldBounds(8192.0);
+
+            IO::TestParserStatus status;
+            WorldReader reader(data, Model::MapFormat::Standard);
+
+            auto world = reader.read(worldBounds, status);
+
+            CHECK(world->childCount() == 2u);
+
+            // NOTE: They are listed in world->children() in file order, not sort index order
+            auto* defaultLayerNode = dynamic_cast<Model::LayerNode*>(world->children().at(0));
+            auto* customLayerNode  = dynamic_cast<Model::LayerNode*>(world->children().at(1));
+
+            REQUIRE(defaultLayerNode != nullptr);
+            REQUIRE(customLayerNode != nullptr);
+
+            auto* groupNode1 = dynamic_cast<Model::GroupNode*>(customLayerNode->children().front());
+            auto* groupNode2 = dynamic_cast<Model::GroupNode*>(defaultLayerNode->children().front());
+
+            REQUIRE(groupNode1 != nullptr);
+            REQUIRE(groupNode2 != nullptr);
+
+            CHECK(customLayerNode->persistentId() == 7u);
```

Comment:
check `defaultLayerNode->persistentId` for completeness

Reply:
done! 6bc44188fe193491265c8c39c3b8e84c3087b90e